src/snowflake/cli/_plugins/streamlit/streamlit_entity.py

```diff
@@ -28,43 +25,140 @@ def __init__(self, *args, **kwargs):
+
+    def action_drop(self, action_ctx: ActionContext, *args, **kwargs):
+        return self._sql_executor.execute_query(self.get_drop_sql())
+
+    def action_execute(
+        self, action_ctx: ActionContext, *args, **kwargs
+    ) -> SnowflakeCursor:
+        return self._sql_executor.execute_query(self.get_execute_sql())
+
+    def action_get_url(
+        self, action_ctx: ActionContext, *args, **kwargs
+    ):  # maybe this should be a property
+        name = self._entity_model.fqn.using_connection(self._conn)
+        return make_snowsight_url(
+            self._conn, f"/#/streamlit-apps/{name.url_identifier}"
         )
 
+    def bundle(self, output_dir: Optional[Path] = None):
+
+        if not output_dir:
+            output_dir = self.root / "output" / self._entity_model.stage
+
+        artifacts = self._entity_model.artifacts
+
+        output_dir.mkdir(parents=True, exist_ok=True)  # type: ignore
+
+        output_files = []
+
+        # This is far from , but will be replaced by bundlemap mappings.
+        for file in artifacts:
+            output_file = output_dir / file.name
+
+            if file.is_file():
+                SecurePath(file).copy(output_file)
+            elif file.is_dir():
+                output_file.mkdir(parents=True, exist_ok=True)
+                SecurePath(file).copy(output_file, dirs_exist_ok=True)
+
+                output_files.append(output_file)
+
+        return output_files
+
+    def action_share(
+        self, action_ctx: ActionContext, to_role: str, *args, **kwargs
+    ) -> SnowflakeCursor:
+        return self._sql_executor.execute_query(self.get_share_sql(to_role))
+
     def get_deploy_sql(
         self,
+        if_not_exists: bool = False,
+        replace: bool = False,
+        from_stage_name: Optional[str] = None,
         artifacts_dir: Optional[Path] = None,
         schema: Optional[str] = None,
+        *args,
+        **kwargs,
     ):
-        entity_id = self.entity_id
-        if artifacts_dir:
-            streamlit_name = f"{schema}.{entity_id}" if schema else entity_id
-            return f"CREATE OR REPLACE STREAMLIT {streamlit_name} FROM '{artifacts_dir}' MAIN_FILE='{self._entity_model.main_file}';"
+        if replace and if_not_exists:
+            raise ClickException("Cannot specify both replace and if_not_exists")
+
+        if replace:
+            query = "CREATE OR REPLACE "
+        elif if_not_exists:
+            query = "CREATE IF NOT EXISTS "
         else:
-            return f"CREATE OR REPLACE STREAMLIT {entity_id} MAIN_FILE='{self._entity_model.main_file}';"
+            query = "CREATE "
+
+        schema_to_use = schema or self._entity_model.fqn.schema
+        query += f"STREAMLIT {self._entity_model.fqn.set_schema(schema_to_use).sql_identifier}"
+
+        if from_stage_name:
+            query += f"\nROOT_LOCATION = '{from_stage_name}'"
+        elif artifacts_dir:
+            query += f"\nFROM = '{artifacts_dir}'"
```

Comment:
I think having the `=` here is invalid syntax ([docs](https://docs.snowflake.com/en/developer-guide/native-apps/adding-streamlit#create-the-streamlit-object-in-the-setup-script))
```suggestion
            query += f"\nFROM '{artifacts_dir}'"
```